games-strategy/opendune/patches/opendune-0.9.patchset

```diff
@@ -805,5 +805,37 @@ index e623ebd..d8489ae 100644
  assert_compile(sizeof( int8 ) == 1);
  assert_compile(sizeof(uint16) == 2);
 -- 
-2.28.0
+2.45.2
+
+
+From 9ec10dd99f95d605e7b72bea3f7dc5c99054b039 Mon Sep 17 00:00:00 2001
+From: Peppersawce <michaelpeppers89@yahoo.it>
+Date: Thu, 26 Sep 2024 23:08:31 +0200
+Subject: Hackfix (with sed) to Opendune using ~/.config to store user data
+
+
+diff --git a/src/file.c b/src/file.c
+index 0b08d60..1f23cb3 100644
+--- a/src/file.c
++++ b/src/file.c
+@@ -42,7 +42,7 @@
+ 
+ #ifdef __HAIKU__
+ static char g_dune_data_dir[B_PATH_NAME_LENGTH] = DUNE_DATA_DIR;
+-static char g_personal_data_dir[B_PATH_NAME_LENGTH] = ".";
++static char g_personal_data_dir[B_PATH_NAME_LENGTH] = "@HAIKUSERSETTINGS@";
+ #else
+ static char g_dune_data_dir[1024] = DUNE_DATA_DIR;
+ static char g_personal_data_dir[1024] = ".";
+@@ -558,7 +558,7 @@ bool File_Init(void)
+ #if defined(__APPLE__)
+ 			snprintf(g_personal_data_dir, sizeof(g_personal_data_dir), "%s/Library/Application Support/OpenDUNE", homedir);
+ #else /* __APPLE__ */
+-			snprintf(g_personal_data_dir, sizeof(g_personal_data_dir), "%s/.config/opendune", homedir);
++			snprintf(g_personal_data_dir, sizeof(g_personal_data_dir), "@HAIKUSERSETTINGS@");
```

Comment:
This line isn't needed because you are already initializing g_personal_data_dir with the same value.

Reply:
Can't read that kind of code ngl, will amend

Reply:
Without that line it saves stuff in ~/.config for some reason

Reply:
If for any reason it'd be better to do it the other way around (changing the value earlier and removing said line) as you suggested at first let me know :)